.github/workflows/publish-on-tag.yml

```diff
@@ -15,6 +15,8 @@ jobs:
         run: npm install
       - name: Build
         run: npm run build
+      - name: Test types.d.ts file is correct
+        run: npm run test-types-file
```

Comment:
There should be no need for this, since PRs should only be merged and releases only created once `main.yml` checks have already passed. This is why we don't run any other checks here as well. This will only slow down releases unnecessarily so let's drop it